src/arrayMethodSort.js

```diff
@@ -4,8 +4,27 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => {
+      if (a + '' < b + '') {
+        return -1;
+      }
+
+      return 1;
+    },
+  ) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length; j++) {
+        if (i !== j && compareFunction(this[i], this[j]) < 0) {
+          const temp = this[j];
+
+          this[j] = this[i];
+          this[i] = temp;
+        }
+      }
+    }
+
+    return this;
```

Comment:
Consider adding a check to see if the array is already sorted before starting the sort process, as an optimization for cases when the array is already in order.